src/sentry/tasks/integrations/github/link_all_repos.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import sentry_sdk
+
+from sentry.models.organization import Organization
+from sentry.plugins.base import bindings
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.integrations.github.pr_comment import RATE_LIMITED_MESSAGE
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(name="sentry.integrations.github.link_all_repos")
+def link_all_repos(integration_id: int, organization_id: int):
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if not integration:
+        logger.error(
+            "github.link_all_repos.integration_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
+        return
+
+    try:
+        organization = Organization.objects.get(id=organization_id)
+    except Organization.DoesNotExist:
+        logger.error(
+            "github.link_all_repos.orgnanization_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr("github.link_all_repos.error", tags={"type": "missing_organization"})
+        return
+
+    installation = integration_service.get_installation(
+        integration=integration, organization_id=organization_id
+    )
+
+    gh_client = installation.get_client()
+
+    try:
+        gh_repositories = gh_client.get_repositories(fetch_max_pages=True)
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr("github.link_all_repos.rate_limited_error")
+            return
+
+        metrics.incr("github.link_all_repos.api_error")
+        raise e
```

Comment:
We could retry the task with the pagination key specified some way if we encounter rate limiting. It might not be the case, that way we don't just create a portion of the repositories.

Reply:
we will have some logic added onto the existing github webhook to create a `Repository` row if we ever find that a repo is mentioned but not installed. would this cover the case you mentioned?